tests/test_web_server.py

```diff
@@ -90,8 +90,9 @@ def handler(request):
     server = yield from raw_test_server(handler, logger=logger)
     cli = yield from test_client(server)
 
-    with pytest.raises(client.ServerDisconnectedError):
-        yield from cli.get('/path/to')
+    resp = yield from cli.get('/path/to')
+    with pytest.raises(client.ClientPayloadError):
```

Comment:
other note. buffering is needed because of http pipelining. but if we decide on #2109 then it may go away.